CMakeLists.txt

```diff
@@ -1,34 +1,28 @@
-cmake_minimum_required (VERSION 2.8.8)
+cmake_minimum_required(VERSION 3.12)
```

Comment:
isnt the min required 3.1 only ?

Reply:
This is intentional as 3.1 doesn't support some features. I haven't figured out the exact version necessary, but it's something around ~ 3.5 or 3.7. I usually stick with the version supported by Travis (… if not to outdated…).

If 3.12 is to new, I can check the docs for what's actually the lowest version.

Reply:
I use 3.10 at home. Am I an old fart ? :(

Reply:
No problem, v3.10 is fine too.

Reply:
3.5 is ubuntu 16.04. Not that old.
What features are you talking about ?

Reply:
It doesn't support gtest features like the imported targets or the improved test discovering (coming soon). More recent version also have greatly improved the doxygen module, a feature worth considering.

Reply:
I see this as a debate of "modernity" vs "compatibility". Both arguments are valuable out of context, but here, I think that compatibility is better.
safe will certainly not evolve into a complex library in which advanced features are needed. Allowing anyone who wants to to compile and run the tests (compatibility) seems more valuable to me than having access to advanced cmake features.
I also think that the cmake_minimum_required should indeed be the minimum required version needed to run the cmake! What's the oldest required version right now ? 3.1 ?
The distro we use in Travis is 16.04, so the cmake required version certainly should be lower than 3.5!

Reply:
3.1 seems to do the trick !

Reply:
Ok, I'll update this PR accordingly.